packages/projection/src/sinks/inMemory/stakePools.ts

```diff
@@ -0,0 +1,38 @@
+import { Cardano, ChainSyncEventType } from '@cardano-sdk/core';
+import { EMPTY } from 'rxjs';
+import { Sink } from '../types';
+import { StakePoolsProjection } from '../../projections';
+import { WithCertificateSource } from '../../operators';
+import { WithInMemoryStore } from './types';
+
+const findOrCreate = ({ store: { stakePools } }: WithInMemoryStore, poolId: Cardano.PoolId) => {
+  let stakePool = stakePools.get(poolId);
+  if (!stakePool) {
+    stakePools.set(poolId, (stakePool = { retirements: [], updates: [] }));
+  }
+  return stakePool;
+};
+
+export const stakePools: Sink<StakePoolsProjection, WithInMemoryStore> = {
+  sink(evt) {
+    if (evt.eventType === ChainSyncEventType.RollForward) {
+      for (const [poolId, poolUpdates] of evt.stakePools.updates) {
+        findOrCreate(evt, poolId).updates.push(...poolUpdates);
+      }
+      for (const [poolId, poolRetirements] of evt.stakePools.retirements) {
+        findOrCreate(evt, poolId).retirements.push(...poolRetirements);
+      }
+    } else {
+      // Delete all updates and retirements >= current tip.
+      // Alternative implementation would be iterating over updates/retirements in the event
```

Comment:
I think it's satisfactory as-is. Better performance this way

Reply:
Removed comment about the alternative 08de91767b3d9401e4e4df4541a442158a5ff403